api/v1alpha1/vaultauth_types.go

```diff
@@ -25,6 +26,28 @@ type VaultAuthConfigKubernetes struct {
 	TokenExpirationSeconds int64 `json:"tokenExpirationSeconds,omitempty"`
 }
 
+// Token represents the token if present in a VaultAuthConfigJwt.
+type Token struct {
+	ValueFrom *corev1.EnvVarSource `json:"valueFrom"`
```

Comment:
[EnvVarSource](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#envvarsource-v1-core) supports more sources than just secrets, so I wonder if we should restrict it to only secrets? It feels a little odd to me to allow tokens to be retrieved from non-secrets.

Perhaps this could just be our own object that takes a secret name and key within the secret?

Reply:
Thanks! I replaced field Token for `TokenSecretKeySelector` with type `*corev1.SecretKeySelector`